### DISCUSSION TITLE
Create a 'safe amperage' setting to which the EVSE can resort when there is an issue with grid or EV meter data

### DISCUSSION THREAD

**Discussion by thmjanssen** [dingo35/SmartEVSE-3.5#135] [Ideas]
Currently, if the EVSE does not receive values for Mains meter, EV meter or Battery, it enters an error state and it stops charging (or at least that's what I think I see happening). It leads to several SPOF's. I was surprised in particular that not receiving EV meter data breaks charging functionality.

Could we get a setting for a configurable amperage to which the EVSE can fallback when there is an issue with data submission? So something like a 'safe-amperage' value, and when there is an issue with receiving data the EVSE behaves like Normal mode but with a restricted amperage per said setting.
If data is being received again, it moves back to it's previous mode and amperage.

**Comment by dingo35** [comment]
That is considered to be unsafe, since you want to guard your fuses (thats why you choose Smart mode or Solar mode).

This is not only my personal opinion, but supported by many in the community (although a few support your position).

The mainstream vision is this would cure the symptom, not the cause. Better invest in a stable infrastructure instead. Why do you have so many disturbances?

**Comment by thmjanssen** [comment]
While I certainly have some reliability improvements to make in my infrastructure, I am trying to make a different point. Running a 6A workload should not trip the main fuse, there are several chargers on the market that work without considering the load on the main fuse. So hence a safe amperage to fallback to.

But the key point is that optional data like the EV meter (especially when not used for billing) and battery current should not stop the charging if you ask me. In particular with the timings being so strict as they are. If I don't submit battery current every 5 seconds, the charger stops. Same for the EV meter data. 

For example. while not ideal, I don't mind the home battery being emptied into my car because a script failed that submits the battery behavior to the EVSE. I do mind walking up to an empty car in the morning because said script failed.

There is now so much logic in the EVSE that depends on several sources for it's proper functioning, I just think it would benefit from some add'l reliability logic that separates the must-have things from the nice-to-have things and influences the functioning accordingly so charging can continue under most circumstances.

**Reply by thmjanssen** [reply]
BTW lack of recent battery current data does not appear to stop charging, but lack of EV meter and Mains Meter data does.

**Reply by thmjanssen** [reply]
https://knowledge.alfen.com/download/attachments/639762449/203130187-ICU_SCG_Eve_Single_NG910%2B_V2.0.pdf

This is the Alfen manual, section 5.6.3 describes the same function in the settings of their Smart-mode equivalent: 

Data timeout from EMS ==> fallback to safe current.
